rocketpy/sensors/sensors.py

```diff
@@ -0,0 +1,371 @@
+        # counter for cases where the sensor is added multiple times in a rocket
+        self._counter += 1
+        if self._counter == len(self.measured_data):
+            self._counter = 0
+
+    @abstractmethod
+    def measure(self, *args, **kwargs):
+        pass
+
+    @abstractmethod
+    def export_measured_data(self):
+        pass
+
+    def quantize(self, value):
+        """
+        Quantize the sensor measurement
+
+        Parameters
+        ----------
+        value : float
+            The value to quantize
+
+        Returns
+        -------
+        float
+            The quantized value
+        """
+        x = min(max(value.x, self.measurement_range[0]), self.measurement_range[1])
+        y = min(max(value.y, self.measurement_range[0]), self.measurement_range[1])
+        z = min(max(value.z, self.measurement_range[0]), self.measurement_range[1])
+        if self.resolution != 0:
+            x = round(x / self.resolution) * self.resolution
+            y = round(y / self.resolution) * self.resolution
+            z = round(z / self.resolution) * self.resolution
+        return Vector([x, y, z])
+
+    def apply_noise(self, value):
+        """
+        Add noise to the sensor measurement
+
+        Parameters
+        ----------
+        value : float
+            The value to add noise to
+
+        Returns
+        -------
+        float
+            The value with added noise
+        """
+        # white noise
+        white_noise = (
+            Vector(
+                [np.random.normal(0, self.noise_variance[i] ** 0.5) for i in range(3)]
+            )
+            & self.noise_density
+        ) * self.sampling_rate**0.5
+
+        # random walk
+        self._random_walk_drift = (
+            self._random_walk_drift
+            + (
+                Vector(
+                    [
+                        np.random.normal(0, self.random_walk_variance[i] ** 0.5)
+                        for i in range(3)
+                    ]
+                )
+                & self.random_walk_density
+            )
+            / self.sampling_rate**0.5
+        )
+
+        # add noise
+        value += white_noise + self._random_walk_drift + self.constant_bias
+
+        return value
+
+    def apply_temperature_drift(self, value):
+        """
+        Apply temperature drift to the sensor measurement
+
+        Parameters
+        ----------
+        value : float
+            The value to apply temperature drift to
+
+        Returns
+        -------
+        float
+            The value with applied temperature drift
+        """
+        # temperature drift
+        value += (self.operating_temperature - 25) * self.temperature_bias
+        # temperature scale factor
+        scale_factor = (
+            Vector([1, 1, 1])
+            + (self.operating_temperature - 25) / 100 * self.temperature_scale_factor
+        )
+        value = value & scale_factor
+
+        return value
```

Comment:
> So you could do something like: `Vector.ones()` instead of `Vector([1, 1, 1])`. (Spoiler: It doesnt make much difference)

I had implemented `Vector.eye()` to be like matlab, but honestly it just made the code harder to read so I left it defining explicitly like this